src/reducers/productReducer.js

```diff
@@ -0,0 +1,88 @@
+import { Actions } from "./Actions";
+import { appendItem, updateQuantity, removeItemFromCartList } from "./utils";
+
+const productReducer = (prevState, action) => {
+  switch (action.type) {
+    case Actions.INITIALIZE_PRODUCTS:
+      return {
+        ...prevState,
+        productList: action.payload,
+      };
+    case Actions.INITIALIZE_WISH:
+      return {
+        ...prevState,
+        wishList: action.payload,
+      };
+    case Actions.INITIALIZE_CART:
```

Comment:
Another suggestion here. Now, this has got to do nothing with functionality. Just a trick to reduce the number of lines in code.  Feel free to ignore this if it seems complex.

Instead of having 3 actions INITIALIZE_PRODUCTS, INITIALIZE_WISH and INITIALIZE_CART which essentially do the same thing, you can have one action here. 

```
case INITIALIZE_ARRAY:
  return {
   ...prevState,
   [payload.arrayName]: payload.data,
 }
```

Here, `arrayName` could be a string variable with value as "wishList", "productList" or "cartList"
Feel free to change the variable names I have used so that they make more sense :)
Good variable names are still something I am working on.